src/core.c/Buf.pm6

```diff
@@ -291,23 +291,34 @@ my role Blob[::T = uint8] does Positional[T] does Stringy is repr('VMArray') is
 
     proto method decode(|) {*}
     multi method decode(Blob:D: $encoding = self.encoding // "utf-8") {
-        nqp::p6box_s(
-          nqp::decode(self, Rakudo::Internals.NORMALIZE_ENCODING($encoding))
-        )
+        if Rakudo::Internals.BUILTIN_ENCODINGS{$encoding} {
```

Comment:
I fail to see the use of `Rakudo::Internals.BUILTIN_ENCODINGS`.  I think the condition could be better written as:

```` raku
if Rakudo::Internals.NORMALIZE_ENCODING($encoding) -> $normalized {
    nqp::p6box_s(nqp::decode(self,$normalized))
}

and the same at similar places in this PR?

Reply:
That was actually my first thought as well!  However, that doesn't work because `NORMALIZE_ENCODING` falls back to returning a lower-cased version of the encoding when passed an unknown encoding:

```raku
say Rakudo::Internals.NORMALIZE_ENCODING('Foo')  # OUTPUT: «foo»
```

(I suppose the thought here is that we're _normalizing_ encodings, not checking whether they're known.)  This means that the if block you suggested would always return `True`.  Of course, we could modify the behavior of `NORMALIZE_ENCODINGS`, but I thought it would be better to suggest the smaller change.

Reply:
https://github.com/rakudo/rakudo/commit/22ea9d903f makes R:I:NORMALIZE_ENCODING return `Nil` rather than a lowercase version of the input.

Reply:
Fixed in 68dbb99, thanks.